csrc/mmdeploy/net/snpe/snpe_net.cpp

```diff
@@ -0,0 +1,257 @@
+  }
+
+  snpe_ =
+      snpeBuilder.setOutputLayers({})
+          .setRuntimeProcessorOrder(runtimeList)
+          .setUseUserSuppliedBuffers(useUserSuppliedBuffers)
+          .setPlatformConfig(platformConfig)
+          .setPerformanceProfile(zdl::DlSystem::PerformanceProfile_t::SUSTAINED_HIGH_PERFORMANCE)
+          .build();
+  return;
+}
+
+void SNPENet::copy_output(const zdl::DlSystem::ITensor* from, Tensor& to) {
+  auto hwc_to_chw = [](const zdl::DlSystem::TensorShape& shape) -> bool {
+    if (shape.rank() != 4 || (shape[1] == 1 && shape[2] > 1 && shape[3] > 1)) {
+      return false;
+    }
+    return true;
+  };
+
+  auto output_shape = from->getShape();
+
+  if (to.size() != from->getSize()) {
+    TensorShape tensor_shape;
+    for (int j = 0; j < output_shape.rank(); ++j) {
+      tensor_shape.push_back(output_shape[j]);
+    }
+
+    if (hwc_to_chw(output_shape)) {
+      auto tmp = output_shape[3];
+      output_shape[3] = output_shape[1];
+      output_shape[1] = tmp;
+    }
+    to.Reshape(tensor_shape);
+  }
+
+  float* pto = to.data<float>();
+
+  if (output_shape.rank() != 4 ||
+      (output_shape[1] == 1 && output_shape[2] > 1 && output_shape[3] > 1)) {
+    // skip [1,1,w>1,h>1] for segmentation task
+    for (auto it = from->cbegin(); it != from->cend(); ++it, ++pto) {
+      *pto = *it;
+    }
+  } else {
+    const int channel = output_shape[1];
+    const int panel = output_shape[2] * output_shape[3];
+
+    int i = 0;
+    // HWC to CHW
+    for (auto it = from->cbegin(); it != from->cend(); ++it, ++i) {
+      int channel_idx = i % channel;
+      int panel_idx = i / channel;
+      pto[channel_idx * panel + panel_idx] = *it;
+    }
+  }
+  return;
+}
+
+void SNPENet::copy_input(const Tensor& from, zdl::DlSystem::ITensor* to) {
+  if (from.size() != to->getSize()) {
+    MMDEPLOY_ERROR("input tensor size not match");
+    return;
+  }
+
+  const float* pfrom = from.data<float>();
+
+  auto input_shape = to->getShape();
+  if (input_shape.rank() == 4) {
+    const int channel = input_shape[3];
+    const int panel = input_shape[1] * input_shape[2];
+
+    int i = 0;
+    // CHW to HWC
+    for (auto it = to->begin(); it != to->end(); ++it, ++i) {
+      int channel_index = i % channel;
+      int panel_index = (i / channel) % panel;
+
+      *it = pfrom[channel_index * panel + panel_index];
+    }
+
+  } else {
+    for (auto it = to->begin(); it != to->end(); ++it, ++pfrom) {
+      *it = *pfrom;
+    }
+  }
+}
+
+Result<void> SNPENet::Init(const Value& args) {
+  auto& context = args["context"];
+  device_ = context["device"].get<Device>();
+  stream_ = context["stream"].get<Stream>();
+  if (!device_.is_host()) {
+    return Status(eNotSupported);
+  }
+
+  auto name = args["name"].get<std::string>();
+  auto model = context["model"].get<Model>();
+  OUTCOME_TRY(auto config, model.GetModelConfig(name));
+
+  container_ = zdl::DlContainer::IDlContainer::open(zdl::DlSystem::String(config.net));
```

Comment:
It is recommend to read model from buffer instead of from file like [this](https://github.com/open-mmlab/mmdeploy/blob/0e1a3aa8c7ce0a0dd494132fd9afb4c06e881913/csrc/mmdeploy/net/ncnn/ncnn_net.cpp#L60) .

Reply:
done.